lib/asn.c

```diff
@@ -0,0 +1,59 @@
+/*
+ * ASN functions
+ *
+ * Copyright 2022 6WIND
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the Free
+ * Software Foundation; either version 2 of the License, or (at your option)
+ * any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; see the file COPYING; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA
+ */
+#include <zebra.h>
+#include "log.h"
+#include "asn.h"
+
+static int strnodot2asn(const char *asstring, as_t *asn)
+{
+	uint32_t local_asn;
+
+	if (!asstring)
+		return 0;
+
+	if (sscanf(asstring, "%u", &local_asn) != 1)
+		return 0;
+
+	if (!asn)
```

Comment:
Can we move this above to return earlier?

Reply:
ok